### DISCUSSION TITLE
clangd parse code very slow in header only  template library nvidia cutlass

### DISCUSSION THREAD

**Discussion by CalebDu** [clangd/clangd#2186] [Q&A]
I use Nvidia Cutlass library to build my program. But I find that clangd parse single code file very slow by using more template class/function, typing take more than ten seconds to parse and give codecompletion in vscode. I coding in remote server so hardware is powerful. And I notice that only 1-2 cpu get 100% usage when clangd parsing. 
How to deal with and speed up?

**Comment by kozoy** [comment]
I failed using cland to parse some complex template header in cutlass, I already add the include path to configuration, but it didn't work.
![image](https://github.com/user-attachments/assets/7afe3fd8-b7b7-4dc0-bde1-45819690a341)
![image](https://github.com/user-attachments/assets/52cffc12-1ea9-466e-9d4c-fe0a4d8d4907)
